tests/integration/cli/tests/run.rs

```diff
@@ -546,3 +546,58 @@ fn run_no_start_wasm_report_error() -> anyhow::Result<()> {
     assert_eq!(result.contains("Can not find any export functions."), true);
     Ok(())
 }
+
+// Test that wasmer can run a complex path
+#[test]
+fn test_wasmer_run_complex_url() -> anyhow::Result<()> {
+    let temp_dir = tempfile::tempdir()?;
+    let path = temp_dir.path().to_path_buf();
+    let root = get_repo_root_path()
+        .unwrap()
+        .join("temp-wasmer-run-complex-url");
+    #[cfg(target_os = "windows")]
+    {
+        // wasmer run used to fail on c:\Users\username\wapm_packages\ ...
+        let root_str = format!("{}", root.display());
+        assert!(
+            root_str.starts_with("c:\\") || root_str.starts_with("C://"),
+            "windows path is not complex enough"
+        );
+    }
+
+    println!(
+        "copying from {} to {}",
+        wasi_test_wasm_path(),
+        root.join("qjs.wasm").display()
+    );
+    std::fs::create_dir_all(&root).unwrap();
+    println!("copying...");
+    std::fs::copy(wasi_test_wasm_path(), root.join("qjs.wasm")).unwrap();
```

Comment:
You don't need to copy or create a temp dir, you can just use the qjs.wasm file normally (with display for windows)
: `wasi_test_wasm_path().display()`

Reply:
ok done